src/main/java/net/craftcitizen/imagemaps/ImageMapDownloadCommand.java

```diff
@@ -9,95 +16,90 @@
         String url = args[2];
-        
+
         if (filename.contains("/") || filename.contains("\\") || filename.contains(":")) {
             MessageUtil.sendMessage(getPlugin(), sender, MessageLevel.WARNING, "Filename contains illegal character.");
             return null;
         }
-        
+
         new LambdaRunnable(() -> download(sender, url, filename)).runTaskAsynchronously(plugin);
         return null;
     }
-    
+
     private void download(CommandSender sender, String input, String filename) {
         try {
             URL srcURL = new URL(input);
-            
+
             if (!srcURL.getProtocol().startsWith("http")) {
                 MessageUtil.sendMessage(getPlugin(), sender, MessageLevel.WARNING, "Download URL is not valid.");
                 return;
             }
-            
+
             URLConnection connection = srcURL.openConnection();
-            
+
             if (!(connection instanceof HttpURLConnection)) {
                 MessageUtil.sendMessage(getPlugin(), sender, MessageLevel.WARNING, "Download URL is not valid.");
                 return;
             }
-            
+
+            connection.setRequestProperty("User-Agent", "ImageMaps/0");
+
             if (((HttpURLConnection) connection).getResponseCode() != 200) {
                 MessageUtil.sendMessage(getPlugin(),
-                                        sender,
-                                        MessageLevel.WARNING,
-                                        String.format("Download failed, HTTP Error code %d.", ((HttpURLConnection) connection).getResponseCode()));
+                        sender,
+                        MessageLevel.WARNING,
+                        String.format("Download failed, HTTP Error code %d.", ((HttpURLConnection) connection).getResponseCode()));
                 return;
             }
-            
-            String mimeType = ((HttpURLConnection) connection).getHeaderField("Content-type");
+
+            String mimeType = connection.getHeaderField("Content-type");
             if (!(mimeType.startsWith("image/"))) {
                 MessageUtil.sendMessage(getPlugin(), sender, MessageLevel.WARNING, String.format("Download is a %s file, not image.", mimeType));
                 return;
             }
-            
-            if (ImageIO.read(srcURL) == null) {
+
+            try (InputStream str = connection.getInputStream()) {
+                BufferedImage image = ImageIO.read(str);
+                File outFile = new File(plugin.getDataFolder(), "images" + File.separatorChar + filename);
+                boolean fileExisted = outFile.exists();
+                ImageIO.write(image, "PNG", outFile);
+                if (fileExisted) {
+                    MessageUtil.sendMessage(getPlugin(), sender, MessageLevel.WARNING, "File already exists, overwriting!");
+                    getPlugin().reloadImage(filename);
+                }
+            } catch (IllegalArgumentException ex) {
```

Comment:
Reinstate `image == null` check. ImageIO#read is specified to return null if it can't read the input as image.

Reply:
Done